WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java

```diff
@@ -33,62 +36,161 @@
+    // so we need to be able to map them to the Post that failed
+    private static final SparseArray<Integer> sPostIdToErrorNotificationId = new SparseArray<>();
+
+    // used to hold notification data for everything (only one outstanding foreground notification
+    // for the live UploadService instance
+    private static NotificationData sNotificationData;
 
     private class NotificationData {
         int notificationId;
-        int notificationErrorId;
         int totalMediaItems;
         int currentMediaItem;
-        float itemProgressSize;
-        Bitmap latestIcon;
+        int totalPostItems;
+        int totalPageItemsIncludedInPostCount;
+        int currentPostItem;
     }
 
     PostUploadNotifier(Context context, UploadService service) {
         // Add the uploader to the notification bar
         mContext = context;
         mService = service;
+        sNotificationData = new NotificationData();
         mNotificationManager = (NotificationManager) SystemServiceFactory.get(mContext,
                 Context.NOTIFICATION_SERVICE);
         mNotificationBuilder = new Notification.Builder(mContext.getApplicationContext());
         mNotificationBuilder.setSmallIcon(android.R.drawable.stat_sys_upload);
     }
 
-    void showForegroundNotificationForPost(@NonNull PostModel post, String message) {
-        mNotificationBuilder.setContentTitle(buildNotificationTitleForPost(post));
-        if (message != null) {
-            mNotificationBuilder.setContentText(message);
+    private void updateForegroundNotification(@Nullable PostModel post) {
+        updateNotificationBuilder(post);
+        doNotify(sNotificationData.notificationId, mNotificationBuilder.build());
+    }
+
+    private void updateNotificationBuilder(@Nullable PostModel post) {
+        // set the Notification's title and prepare the Notifications message text, i.e. "1/3 Posts, 4/17 media items"
+        if (sNotificationData.totalMediaItems > 0 && sNotificationData.totalPostItems == 0) {
+            // only media items are being uploaded
+            // check if special case for ONE media item
+            if (sNotificationData.totalMediaItems == 1) {
+                mNotificationBuilder.setContentTitle(buildNotificationTitleForMedia());
+                mNotificationBuilder.setContentText(buildNotificationSubtitleForMedia());
+            } else {
+                mNotificationBuilder.setContentTitle(buildNotificationTitleForMixedContent());
+                mNotificationBuilder.setContentText(buildNotificationSubtitleForMedia());
+            }
+
+        } else if (sNotificationData.totalMediaItems == 0 && sNotificationData.totalPostItems > 0) {
+            // only Post / Pages are being uploaded
+            // check if special case for ONE Post
+            if (sNotificationData.totalPostItems == 1) {
+                mNotificationBuilder.setContentTitle(buildNotificationTitleForPost(post));
```

Comment:
Since `post` is `@Nullable` should we be null checking it here?

Reply:
good call, I handled it in a slightly different way within `buildNotificationTitleForPost` and `buildNotificationSubtitleForPost` in 90b42df